src/lib.rs

```diff
@@ -224,7 +279,27 @@ pub trait Resampler<T>: Send {
 pub trait VecResampler<T>: Send {
     /// Resample a chunk of audio.
     /// Input and output data is stored in vectors, where each element contains a vector with all samples for a single channel.
-    fn process(&mut self, wave_in: &[Vec<T>]) -> ResampleResult<Vec<Vec<T>>>;
+    fn process(
+        &mut self,
+        wave_in: &[Vec<T>],
+        active_channels_mask: Option<&[bool]>,
+    ) -> ResampleResult<Vec<Vec<T>>>;
+
+    /// Resample a chunk of audio, into a pre-allocated output buffer.
+    /// Input and output data is stored in vectors, where each element contains a vector with all samples for a single channel.
+    fn process_into_buffer(
+        &mut self,
+        wave_in: &[Vec<T>],
+        wave_out: &mut [Vec<T>],
+        active_channels_mask: Option<&[bool]>,
+    ) -> ResampleResult<()>;
+
+    /// Convenience method for allocating an output buffer suitable for use with `process_into_buffer`.
+    fn allocate_output_buffer(&self) -> Vec<Vec<T>>;
+
+    /// Get the max output size the resampler can give, as (channels, frames).
+    /// Note that when adjusting the ratio of an asynchronous resampler, the maximum size can change.
```

Comment:
I think it would be more maintainable to link to the Resampler method documentation than repeat the text for VecResampler.